packages/smooth_app/lib/temp/matched_product.dart

```diff
@@ -0,0 +1,80 @@
+// Package imports:
+import 'package:openfoodfacts/model/Attribute.dart';
+import 'package:openfoodfacts/model/AttributeGroup.dart';
+import 'package:openfoodfacts/model/Product.dart';
+
+// Project imports:
+import 'package:smooth_app/temp/preference_importance.dart';
+import 'package:smooth_app/temp/product_preferences_manager.dart';
+
+/// Match and score of a [Product] vs. Preferences
+///
+/// cf. https://github.com/openfoodfacts/smooth-app/issues/39
+class MatchedProduct {
+  MatchedProduct(
+    this.product,
+    final ProductPreferencesManager productPreferencesManager,
+  ) {
+    final List<AttributeGroup> attributeGroups = product.attributeGroups;
+    if (attributeGroups == null) {
+      _status = null;
+      return;
+    }
+    for (final AttributeGroup group in attributeGroups) {
+      for (final Attribute attribute in group.attributes) {
+        final PreferenceImportance preferenceImportance =
+            productPreferencesManager.getPreferenceImportanceFromImportanceId(
+          productPreferencesManager.getImportanceIdForAttributeId(
+            attribute.id,
+          ),
+        );
+        final String importanceId = preferenceImportance.id;
+        final int factor = preferenceImportance.factor ?? 0;
+        final int minimalMatch = preferenceImportance.minimalMatch;
+        if (importanceId == null || factor == 0) {
+          _debug += '${attribute.id} $importanceId\n';
+        } else {
+          if (attribute.status == _UNKNOWN_STATUS) {
+            if (_status ?? false) {
+              _status = null;
+            }
+          } else {
+            _debug +=
+                '${attribute.id} $importanceId - match: ${attribute.match}\n';
+            _score += attribute.match * factor;
+            if (minimalMatch != null && attribute.match <= minimalMatch) {
+              _status = false;
+            }
+          }
+        }
+      }
+    }
+  }
+
+  final Product product;
+  double _score = 0;
+  bool _status = true;
+  String _debug = '';
+
+  double get score => _score;
+  bool get status => _status;
+  String get debug => _debug;
+
+  static const String _UNKNOWN_STATUS = 'unknown';
```

Comment:
Those strings are not necessarily used for matching, we probably should put them in the Attributes class, what do you think?

Why _UNKNOWN_STATUS and KNOWN_STATUS without a leading underscore?

Reply:
You're right, they belong in `Attribute`.
The leading underscore means "private", and `KNOW_STATUS` is used in another class.
Anyway it' going to be `STATUS_KNOWN` and `STATUS_UNKNOWN` now, both being used from other classes.
I'll fix that today.